libs/langchain/langchain/callbacks/file.py

```diff
@@ -5,15 +5,27 @@
 from langchain.schema import AgentAction, AgentFinish
 from langchain.utils.input import print_text
 
+class FileOpenError(Exception):
```

Comment:
@aashishsaini75 thanks for the change. I would prefer not implementing a custom exception without a more complete spec / standard on exceptions within langchain; 

Two potential issues:

* I'm unclear on whether this exception does much except for adding `mode` into the error message
* It feels like the exception should maintain python exception hierarchy and still be an instance of an OSError

Reply:
@eyurtsev, You're absolutely right. This change won't close the open file descriptor